doc/holochain_101/src/zome/read_and_write.md

```diff
@@ -1 +1,154 @@
+Despite this, one must of course still be able to persist data, and read back persisted data, and all of that happens in relation to the local source chain.
+
+With Holochain, there are two aspects that combine to form each "element" in this chain-like log: headers, and entries.
+
+### Headers
+
+The fundamental chain-like aspect of Holochain is driven by "headers". They are a minimal data structure containing only a handful of values, but are sufficient to supply Holochain with verifiable data integrity.
+
+Fundamental to creating the "chain", each header contains a reference to the "last" header. There are a couple aspects to this:
+1. The cryptographic hash (its data "fingerprint") for a header is determined
+2. The hash of the previously written header is used as a fixed-length input for the next header, thus influencing the hash of that new header (since any change to contents change a hash)
+3. The only time this does not apply is for the very first header, for which there is no previous header hash, and a value of `0` is used
+
+Other data contained in the headers includes (but isn't limited to):
+- the public key/address, and the cryptographically signed signature of the author
+- a timestamp conveying the moment of creation of the header
+- the hash of the "entry" being created
+- the "entry type" of the "entry" being created
+
+The meaning and role of entries is covered next.
+
+### Entries
+
+Entries are elements of data which get persisted to the same storage as Headers, yet have a different nature. They represent the substance of the data that is of interest to the end users of Holochain, storing records such as transactions, social media posts, or whatever else app developers come up with. The hash of each entry is stored and referenced in its corresponding header. Every local source chain should have the exact same number of Entries and Headers.
+
+All Entries have a type. Entries can be "system" Entries, or "app" Entries. System Entries are Entries integral to the proper functioning of Holochain. Importantly, there are system Entries written to each local source chain, at the time of its initialization, or "genesis".
+
+#### System Entries
+These include firstly, the DNA itself. The DNA is written as the first entry into each local source chain. Importantly, this is what gives everyone totaly confidence in everyone elses chains: the fact that you all know you started out by running the same code (otherwise, the hash of the initial DNA entry would be different). There will be more discussion on this later.
+
+The second Entry to be written to the chain is information about the agent using this source chain. That includes their "public key" which is also their address on the network.
+
+There are also entries created relating to the "Capabilities" features of Holochain, but those will be covered elsewhere.
+
```

Comment:
What other system entry types are there? Do we want to talk about ones that don't exist yet, like migration entries?

Reply:
according to the core code the system entry types are:
```rust
    Dna,
    AgentId,
    Deletion,
    LinkAdd,
    LinkRemove,
    LinkList,
    ChainHeader,
    ChainMigrate,
    CapTokenGrant,
    CapToken,
```

Reply:
I did review this list, but yeah knowing how much or how little to attend to it here was a challenge. I like the way you bring it in